_strtok.c

```diff
@@ -0,0 +1,68 @@
+#include "main.h"
+
+/** _strtok - tokenize @str
+ * @str: pointer to a string
+ *
+ * Return: string
+ */
+
+char *_strtok(char *str)
+{
+	static char *pos;
+	char *ret, q = '\0';
+	int i = 0;
+
+	if (str != NULL)
+		ret = str;
+	else
+	{
+		if (pos == NULL || *pos == '\0')
+			return (NULL);
+		ret = pos;
+	}
+
+
+	while (ret[i] != '\0')
+	{
+		if (ret[i] == ' ')
+		{
+			ret[i] = '\0';
+			while (ret[i + 1] == ' ')
+			{
+				i++;
+				if (ret[i] == '\0')
+					return (ret);
+			}
+			pos = &ret[i + 1];
+			break;
+		}
+		if (ret[i] == '"')
+			q = '"';
+		else if (ret[i] == '\'')
+			q = '\'';
+
+		if (q != '\0')
+		{
+			ret = &ret[i + 1];
+			while (ret[i] != q)
```

Comment:
won't this will cause some error, `ret[i]` will be equivalent to `ret[i + 1 + i]`, because u updated ret to `&ret[i + 1]`